### PR TITLE
[AccountConnection] Remove se23 conditional logic

### LINKED ISSUE #9909: [AccountConnection] Remove conditional se23 logic

### PR DESCRIPTION
**Pull request by laurkim**
### WHY are these changes introduced?

Resolves #9909.

### WHAT is this pull request doing?

Removes se23 beta conditional logic for `AccountConnection`.

### How to 🎩

[Storybook](https://5d559397bae39100201eedc1-ljctfsyeaf.chromatic.com/?path=/story/all-components-accountconnection--all)

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by laurkim** [comment]
fyi I only accepted the 3 changes in [Chromatic](https://shopify.chromatic.com/build?appId=5d559397bae39100201eedc1&number=18986) relating to AccountConnection specifically

**Review by sam-b-rose — APPROVED** [review]
👏 💯

File: .changeset/tall-chicken-repeat.md [hunk 1]
```diff
@@ -2,4 +2,4 @@
 '@shopify/polaris': major
 ---
 
-Removed Summer Editions experimental styles and code for the following components: Avatar
+Removed Summer Editions experimental styles and code for the following components: `Avatar`, `AccountConnection`
```
Comment (sophschneider) [inline .changeset/tall-chicken-repeat.md:5]: Good idea!! 💗

**Review by sophschneider — APPROVED** [review]

File: .changeset/tall-chicken-repeat.md (hunk 1 above)
Comment (laurkim) [reply .changeset/tall-chicken-repeat.md:5]: Shoutout to @samrose3 for the idea 🙌

**Pull request merged**